Content.Server/Chemistry/EntitySystems/ChemistrySystem.Injector.cs

```diff
@@ -33,6 +42,47 @@ private void InitializeInjector()
         SubscribeLocalEvent<InjectionCancelledEvent>(OnInjectionCancelled);
     }
 
+    private void AddSetTransferVerbs(EntityUid uid, InjectorComponent component, GetVerbsEvent<AlternativeVerb> args)
+        {
+            if (!args.CanAccess || !args.CanInteract || args.Hands == null)
+                return;
+
+            if (!EntityManager.TryGetComponent<ActorComponent?>(args.User, out var actor))
+                return;
+
+            // Custom transfer verb
+            AlternativeVerb custom = new();
+            custom.Text = Loc.GetString("comp-solution-transfer-verb-custom-amount");
+            custom.Category = VerbCategory.SetTransferAmount;
+            custom.Act = () => component.UserInterface?.Open(actor.PlayerSession);
+            custom.Priority = 1;
+            args.Verbs.Add(custom);
+
+            // Add specific transfer verbs according to the container's size
+            var priority = 0;
+            foreach (var amount in DefaultTransferAmounts)
+            {
+                if ( amount < component.MinimumTransferAmount.Int() || amount > component.MaximumTransferAmount.Int())
+                    continue;
+
+                AlternativeVerb verb = new();
+                verb.Text = Loc.GetString("comp-solution-transfer-verb-amount", ("amount", amount));
+                verb.Category = VerbCategory.SetTransferAmount;
+                verb.Act = () =>
+                {
+                    component.TransferAmount = FixedPoint2.New(amount);
+                    args.User.PopupMessage(Loc.GetString("comp-solution-transfer-set-amount", ("amount", amount)));
```

Comment:
Maybe make the delay change based on the amount transferred? Something like 1 second per 5u? Still faster than multiple injections, but gives the target a bit more time.

Reply:
Sounds good. Done.